src/diagnosers/12-dnsrecords.py

```diff
@@ -68,7 +69,9 @@ def check_domain(self, domain, is_main_domain):
             return
 
         base_dns_zone = _get_dns_zone_for_domain(domain)
-        basename = domain.replace(base_dns_zone, "").rstrip(".") or "@"
+        basename = tldextract.extract(domain).domain
```

Comment:
Are you sure the goal is to remove the TLD ?

If i understand the goal of `_get_dns_zone_for_domain` is to get the DNS zone (so not the tld), the test seems confirm the semantic: https://github.com/YunoHost/yunohost/blob/140e50253fac0d3c9aa6fcab9e392a462c914e98/src/tests/test_dns.py#L30

So `basename` should be the part at the left in the DNS configuration.

Reply:
So i confirm there is 2 issues: the way we do the replace (that you have spotted) and the 

```
# yunohost tools shell
Warning: The ldap interface is available in this context
Warning: You don't have IPython installed, consider installing it as it is way better than the standard shell.
Warning: Falling back on the standard shell.
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from yunohost.dns import _get_dns_zone_for_domain
>>> _get_dns_zone_for_domain("phillipludewig.de")
'de'
```

For me the result should be "phillipludewig.de"

Reply:
I note that the domain name `philipludewig.de` don't return a NS record and is not reserved. So the bug appears when the domain is not recorded OR if there is no NS record configured.

Reply:
btw your fix, when its merged into the next release, will enable me to use my yunohost instance again properly ^^
so I am really grateful for your support. :D